htdocs/js/apps/DragNDrop/dragndrop.js

```diff
@@ -0,0 +1,140 @@
+(function() {
+	class DragNDropBucket {
+		constructor(pgData) {
+			this.answerInputId = pgData['answerInputId'];
+			this.bucketId = pgData['bucketId'];
+			this.label = pgData['label'] || '';
+			this.removable = pgData['removable'];
+			this.bucketPool = $('.bucket_pool[data-ans="' + this.answerInputId + '"]').first()[0];
+
+			var $bucketPool = $(this.bucketPool);
+			var $newBucket = this._newBucket(
+				this.bucketId,
+				this.label,
+				this.removable,
+				$bucketPool.find('.hidden.past_answers.bucket[data-bucket-id="' + this.bucketId + '"]')
+			);
+			$bucketPool.append($newBucket);
+
+			var el = this;
+			$newBucket.find('.dd').nestable({
+				group: el.answerInputId,
+				maxDepth: 1,
+				scroll: true,
+				callback: function() {el._nestableUpdate();}
+			});
+			this._nestableUpdate();
+			this._ddUpdate();
+		}
+
+		_newBucket(bucketId, label, removable, $bucketHtmlElement) {
+			var $bucketPool = $(this.bucketPool);
+			var $newBucket = $('<div id="nestable-' + bucketId + '-container" class="dd-container"></div>');
+
+			$newBucket.attr('data-bucket-id', bucketId);
+
+			$newBucket.append($('<div class="nestable-label">' + label + '</div>'));
+			$newBucket.append($('<div class="dd" data-bucket-id="' + bucketId + '"></div>'));
+
+			if (removable != 0) {
+				$newBucket.append($('<a class="btn remove_bucket">Remove</a>'));
+			}
+
+			if ($bucketHtmlElement.find('ol.answer li').length) {
+				var $ddList = $('<ol class="dd-list"></ol>');
+				$bucketHtmlElement.find('ol.answer li').each(function(index) {
+					var $item = $('<li><div class="dd-handle">' + $(this).html() + '</div></li>');
+					$item.addClass('dd-item').attr('data-shuffled-index', $(this).attr('data-shuffled-index'));
+					$ddList.append($item);
+				});
+				$newBucket.find('.dd').first().append($ddList);
+			}
+			$newBucket.css('background-color', 'hsla(' + ((100 + (bucketId)*100) % 360) + ', 40%, 90%, 1)');
+			return $newBucket;
+		}
+
+		_nestableUpdate(e) {
+			var buckets = [];
+			$(this.bucketPool).find('.dd').each(function() {
+				var list = [];
+				$(this).find('li.dd-item').each(function() {
+					list.push($(this).attr('data-shuffled-index'));
+				});
+				if (list.length) {
+					buckets.push('(' + list.join(",") + ')');
+				} else {
+					buckets.push('(-1)');
+				}
+			});
+
+			$("#" +  this.answerInputId).val(buckets.join(","));
+		}
+
+		_ddUpdate() {
+			var answerInputId = this.answerInputId;
+			var $bucketPool = $('.bucket_pool[data-ans="' + answerInputId + '"]').first();
+			var el = this;
+			$(function() {
+				$bucketPool.parent().find('.add_bucket').off();
+				$bucketPool.parent().find('.add_bucket').click(function() {
```

Comment:
The `click` function is deprecated.  This should be changed to
```suggestion
				$bucketPool.parent().find('.add_bucket').on('click', function() {
```
This should be done everywhere in this file that the `click` function is used.